### PR TITLE
London_ITP_Jan|Aung_Ye_Kyaw |Module-structuring-and-testing-data/coursework/sprint-2

### PR DESCRIPTION
**Pull request by sarawone**
Hi,

I hope you have a good day!

Kindly check my answers & code for sprint 2- course work. 
Please let me know for any requirements and provide suggestions.

Thanks for your help

Thanks & Regards,
Aung Ye Kyaw

### CODE REVIEW

File: Sprint-2/5-stretch-extend/format-time.js [hunk 1]
```diff
@@ -7,19 +7,26 @@ function formatAs12HourClock(time) {
   if (hours > 12) {
     return `${hours - 12}:00 pm`;
   }
+  if (hours==24)   // while user input is 24 it will show the result into 12 am .
```
Comment (suziebrown) [inline Sprint-2/5-stretch-extend/format-time.js:10]: If this is a special case, you could add a test specifically testing it

File: Sprint-2/1-key-errors/2.js [hunk 2]
```diff
@@ -1,20 +1,21 @@
-
-// Predict and explain first BEFORE you run any code...
-
-// this function should square any number but instead we're going to get an error
-
-// =============> write your prediction of the error here
```
Comment (suziebrown) [inline Sprint-2/1-key-errors/2.js:6]: Don't forget to make a prediction before testing the code

**Review by suziebrown — CHANGES_REQUESTED** [review]

File: Sprint-1/1-key-exercises/2-initials.js [hunk 3]
```diff
@@ -5,7 +5,8 @@ let lastName = "Johnson";
 // Declare a variable called initials that stores the first character of each string.
 // This should produce the string "CKJ", but you must not write the characters C, K, or J in the code of your solution.
 
-let initials = ``;
+let initials = `${firstName.charAt(0)+middleName.charAt(0)+lastName.charAt(0)}`;
```
[the remark this answers is not in the document]
Comment (sarawone) [reply Sprint-1/1-key-exercises/2-initials.js:8]: This is for sprint 1 . I have corrected this one on another PR related to this . 
Thank you .

File: Sprint-1/1-key-exercises/3-paths.js [hunk 4]
```diff
@@ -17,7 +17,9 @@ console.log(`The base part of ${filePath} is ${base}`);
 // Create a variable to store the dir part of the filePath variable
 // Create a variable to store the ext part of the variable
 
-const dir = ;
-const ext = ;
+const dir = filePath.slice(0,lastSlashIndex+1);
+console.log (`The dir part of the file path ${dir}`);
+const ext = filePath.slice(-3);
```
[the remark this answers is not in the document]
Comment (sarawone) [reply Sprint-1/1-key-exercises/3-paths.js:22]: I have corrected this one on another PR. Thank you .

File: Sprint-2/1-key-errors/1.js [hunk 5]
```diff
@@ -1,20 +1,28 @@
...
-console.log(decimalNumber);
+let number = convertToPercentage(0.5);
+console.log(`The result is ${number}`);
 
 // =============> write your explanation here
 
 // Finally, correct the code to fix the problem
-// =============> write your new code here
+// =============> 
+  //function convertToPercentage(decimalNumber)
```
[the remark this answers is not in the document]
Comment (sarawone) [reply Sprint-2/1-key-errors/1.js:22]: I have changed the structure . Please check and let me know . I hope it is the way you expected . Thank you

File: Sprint-2/1-key-errors/1.js [hunk 6]
```diff
@@ -1,20 +1,28 @@
...
   const percentage = `${decimalNumber * 100}%`;
 
   return percentage;
 }
-
-console.log(decimalNumber);
+let number = convertToPercentage(0.5);
+console.log(`The result is ${number}`);
 
 // =============> write your explanation here
```
[the remark this answers is not in the document]
Comment (sarawone) [reply Sprint-2/1-key-errors/1.js:16]: I added the explanation . Please check . Thank you.

File: Sprint-2/3-mandatory-implement/3-to-pounds.js [hunk 7]
```diff
@@ -4,3 +4,31 @@
...
+  .substring(paddedPenceNumberString.length - 2)
+  .padEnd(2, "0");
+
+  result = pounds + "." + pence;
+  return result;
+}
+
+
+change =  toPounds("499p");
+console.log(`The result is ${change}`);
```
[the remark this answers is not in the document]
Comment (sarawone) [reply Sprint-2/3-mandatory-implement/3-to-pounds.js:35]: I am sorry, I did not get your point . I have changed the input to 399p in function as I understand .
please kindly check and let me know if I am on the wrong track again . Thank you .

File: Sprint-2/4-mandatory-interpret/time-format.js [hunk 8]
```diff
@@ -11,24 +12,26 @@ function formatTimeDisplay(seconds) {
...
 
 // Call formatTimeDisplay with an input of 61, now answer the following:
 
 // b) What is the value assigned to num when pad is called for the first time?
-// =============> write your answer here
-
+// =============> 0
 // c) What is the return value of pad is called for the first time?
-// =============> write your answer here
+// =============> got an error TypeError: num.toString(...).padStart is not a function at the Visualiser
```
[the remark this answers is not in the document]
Comment (sarawone) [reply Sprint-2/4-mandatory-interpret/time-format.js:31]: I will do that . Thank you.

File: Sprint-2/5-stretch-extend/format-time.js (hunk 1 above)
Comment (sarawone) [reply Sprint-2/5-stretch-extend/format-time.js:10]: I have updated the code . Please check and let me know . Thank you .

File: Sprint-2/1-key-errors/2.js (hunk 2 above)
Comment (sarawone) [reply Sprint-2/1-key-errors/2.js:6]: I added the comments . Please check and let me know . Thank you.

File: Sprint-2/4-mandatory-interpret/time-format.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (sarawone) [reply Sprint-2/4-mandatory-interpret/time-format.js:31]: I have update the answer please check ...

File: Sprint-2/5-stretch-extend/format-time.js (hunk 1 above)
Comment (suziebrown) [reply Sprint-2/5-stretch-extend/format-time.js:10]: It's good to test this edge case (midnight) so well done for adding the test.
I think maybe you forgot how 24-hour clock works. At midnight, it will show as `00:00` on a 24-hour clock, and in 12-hour it should be formatted as `12:00 am`.
Can you update the test (and the code if needed) so that it works for midnight (input = `00:00`)?

File: Sprint-2/3-mandatory-implement/3-to-pounds.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (sarawone) [reply Sprint-2/3-mandatory-implement/3-to-pounds.js:35]: Oh I got it :) . I have added the £ at the front . Kindly check . Thank you.

File: Sprint-2/5-stretch-extend/format-time.js (hunk 1 above)
Comment (sarawone) [reply Sprint-2/5-stretch-extend/format-time.js:10]: I have updated the code . Kindly check and let me know . Thank you.

**Review by Haseeb-Khalil — DISMISSED** [review]
Looks Good

File: Sprint-2/5-stretch-extend/format-time.js (hunk 1 above)
Comment (suziebrown) [reply Sprint-2/5-stretch-extend/format-time.js]: It should be `if (hours == 0)` because on a 24-hour clock midnight shows as 00:00, not 24:00
